### PR TITLE
fix(spark): Rework spark images to build on top of java-base image

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

This fixes the missing tzdata-java package in 0.0.0-dev versions
Smoke test passed locally

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: pyspark-k8s/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,8 @@
 # syntax=docker/dockerfile:1
-FROM stackable/image/vector
+FROM stackable/image/java-base
 
 ARG PRODUCT
 ARG PYTHON
-ARG JAVA
```
Comment (adwk67) [inline pyspark-k8s/Dockerfile:6]: should this be JAVA_BASE?

**Review by razvan — APPROVED** [review]
lgtm

File: pyspark-k8s/Dockerfile (hunk 1 above)
Comment (adwk67) [reply pyspark-k8s/Dockerfile:6]: sorry, read that wrong :)

Comment (sbernauer) [reply pyspark-k8s/Dockerfile:6]: Yeah, that *somehow* works ;)

**Comment by siegfriedweber** [comment]
All tests passed on my local machine with the exception of the logging tests but that was expected because the tests are not adapted yet to support the new Vector version.

**Pull request merged**